flaskbb/app.py

```diff
@@ -324,63 +328,80 @@ def get_locale():
 
 def configure_logging(app):
     """Configures logging."""
+    if app.config.get('USE_DEFAULT_LOGGING'):
+        configure_default_logging(app)
+
+    if app.config.get('LOG_CONF_FILE'):
+        logging.config.fileConfig(
+            app.config['LOG_CONF_FILE'], disable_existing_loggers=False
+        )
+
+    if app.config["SQLALCHEMY_ECHO"]:
+        # Ref: http://stackoverflow.com/a/8428546
+        @event.listens_for(Engine, "before_cursor_execute")
+        def before_cursor_execute(
+                conn, cursor, statement, parameters, context, executemany
+        ):
+            conn.info.setdefault('query_start_time', []).append(time.time())
 
-    logs_folder = os.path.join(app.root_path, os.pardir, "logs")
+        @event.listens_for(Engine, "after_cursor_execute")
+        def after_cursor_execute(
+                conn, cursor, statement, parameters, context, executemany
+        ):
+            total = time.time() - conn.info['query_start_time'].pop(-1)
+            app.logger.debug("Total Time: %f", total)
+
+
+def configure_default_logging(app):
     from logging.handlers import SMTPHandler
+
+    logs_folder = app.config.get('LOG_PATH')
+
+    if logs_folder is None:
+        logs_folder = os.path.join(app.root_path, os.pardir, "logs")
+
+    default_log_format = '%(asctime)s %(levelname)s: %(message)s [in %(pathname)s:%(lineno)d]'
+
     formatter = logging.Formatter(
-        '%(asctime)s %(levelname)s: %(message)s '
-        '[in %(pathname)s:%(lineno)d]')
+        app.config.get('LOG_FORMAT') or default_log_format
```

Comment:
The `dict.get(...) or ...` pattern prevents things like the key exists but it's set to None or ''